cypress/integration/networking.spec.ts

```diff
@@ -3,49 +3,165 @@ import * as path from "path";
+          commands.pick("create-project-button").click();
 
-        commands
-          .pick("project-screen", "header")
-          .contains("new-fancy-project")
-          .should("exist");
+          commands
+            .pick("project-screen", "header")
+            .contains("new-fancy-project")
+            .should("exist");
 
-        commands.pick("project-screen", "header", "urn").then(el => {
-          const urn = el.attr("title");
-          if (!urn) {
-            throw "Could not find URN";
-          }
+          commands.pick("project-screen", "header", "urn").then(el => {
+            const urn = el.attr("title");
+            if (!urn) {
+              throw new Error("Could not find URN");
+            }
 
-          nodeManager.asNode(node2);
-          commands.pick("sidebar", "search").click();
-          commands.pick("search-input").type(urn || "");
-          commands.pick("follow-toggle").click();
+            nodeManager.asNode(contributorNode);
+            cy.log("navigate to the 'Following' tab");
+            commands.pick("Following").click();
+            commands
+              .pick("following-tab")
+              .contains("Still looking")
+              .should("not.exist");
+
+            commands.pick("sidebar", "search").click();
+            commands.pick("search-input").type(urn || "");
+            commands.pick("follow-toggle").click();
+
+            cy.log("project is briefly shown in the waitingroom");
+            commands
+              .pick("following-tab")
+              .contains("Still looking")
+              .should("exist");
+            commands
+              .pick("following-tab")
+              .contains(urn.replace("rad:git:", ""))
+              .should("exist");
+
+            // TODO: check that notifications are shown
+          });
+
+          cy.log("project moved out of the waiting area and is available");
           commands
             .pick("following-tab", "project-list-entry-new-fancy-project.xyz")
-            .should("exist");
-        });
-      });
+            .click();
+
+          cy.log("the maintainer shows up in the peer selector");
+          commands.pick("peer-selector").contains("rudolfs").should("exist");
+          commands.pick("peer-selector").contains("maintainer").should("exist");
+          commands.pick("peer-selector").click();
+
+          cy.log("the current user does not show up in the peer selector");
+          commands
+            .pick("peer-dropdown-container")
+            .contains("abbey")
+            .should("not.exist");
+
+          cy.log("add contributor remote on maintainer's node");
+          commands.pick("sidebar", "profile").click();
+          commands.pick("peer-id").then(el => {
+            const peerId = el.attr("title");
+            if (!peerId) {
+              throw new Error("Could not find PeerId");
+            }
+            const shortenedPeerId = peerId.slice(0, 7);
```

Comment:
Instead of getting the peer ID from the UI we could add it to `NodeSession` and return it from the node manager. This would reduce the complexity of the code drastically.

Reply:
I also thought about this initially, but then decided against it thinking it would be good to simulate exactly what a user is doing. I guess picking the IDs from the UI is already covered elsewhere.

https://github.com/radicle-dev/radicle-upstream/pull/1559/commits/e98a7a1d411eb16112ab59a2beb54a65a1ed8d92